File: Easy/283 - move-zeroes/Solution.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        int i=0;
        while(i<n&&nums[i]!=0)i++;
        for(int j=i+1;j<n;j++){
            if(nums[j]==0)
                continue;
            nums[i]^=nums[j];
            nums[j]^=nums[i];
            nums[i]^=nums[j];
            i++;
        }
    }
};

File: easy/283 - move-zeroes/Solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int numsSize=nums.length;
        for(int i= numsSize-1;i>=0;i--){
        if(nums[i]==0){
            for(int j=i+1;j<numsSize;j++){
                nums[j-1]=nums[j];
            }
            nums[numsSize-1]=0;
        }
        }
    }
}